pkg/nacler/nacler.go

```diff
@@ -54,3 +58,21 @@ func (n *Nacler) Open(cipherText []byte) ([]byte, error) {
 	}
 	return decrypted, nil
 }
+
+func (n *Nacler) cachedSharedKey() ([32]byte, error) {
```

Comment:
Does it make sense to call this in `New` instead of inline with Seal or Open?

Reply:
yea, that is cleaner and then we don't need mutex